test/golden/artman_golden_test.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Artman golden tests.
+
+It runs GAPIC generation against the golden input data, can then compare the
+output layout with the expected output. This helps to capture unexpected GAPIC
+output layout change caused by either by artman, toolkit or other runtime
+dependency.
+"""
+
+import os
+import subprocess
+
+import pytest
+
+def test_library_example(clone_googleapis_for_testing=False):
+    golden_dir = os.path.dirname(os.path.realpath(__file__))
+    if clone_googleapis_for_testing:
```

Comment:
This is a little surprising. It tethers _the clone_ to the test itself, when the clone is actually setup.

What I would have expected would have been for there to be a `--googleapis` argument with _a path_ (rather than a boolean); skip the test if it is not present, and use that path if it is. Then, the `git clone` could have been moved up to `.circleci/config.yml`.

This would allow you, in local testing, to point to an _existing_ clone. There are lots of reasons to want this; pretend, for example, that you want to check against `googleapis-private`, or a non-master branch.